src/devices/Card/CreditCard/CreditCard.cs

```diff
@@ -0,0 +1,497 @@
+            {
+                if (ret == 2)
+                {
+                    // It's an error, process it
+                    var err = new ProcessError(received.Slice(0, 2));
+                    return err.ErrorType;
+                }
+
+                FillTagList(Tags, received.Slice(0, ret - 3));
+
+                return ErrorType.ProcessCompletedNormal;
+            }
+            return ErrorType.Unknown;
+        }
+
+        private void FillTagList(List<Tag> theTags, ReadOnlySpan<byte> span, ushort parent = 0x00)
+        {
+            var elem = new BerSplitter(span);
+            foreach (var tag in elem.Tags)
+            {
+                // If it is a template or composed, then we need to split it
+                if ((TagList.Tags.Where(m => m.TagNumber == tag.TagNumber).FirstOrDefault()?.IsTemplate == true) || tag.IsConstructed)
+                {
+                    if (tag.Tags == null)
+                        tag.Tags = new List<Tag>();
+                    FillTagList(tag.Tags, tag.Data, tag.TagNumber);
+                }
+
+                // Data Object Lists are special and not BER encoded, they have only the tag number encoded
+                // Like for the traditional tags but the next element is a single byte indicating the size
+                // Of the object
+                if (TagList.Tags.Where(m => m.TagNumber == tag.TagNumber).FirstOrDefault()?.IsDol == true)
+                {
+                    if (tag.Tags == null)
+                        tag.Tags = new List<Tag>();
+
+                    var dolList = tag.Data;
+                    int index = 0;
+                    while (index < dolList.Length)
+                    {
+                        //Decode mono dimention (so 1 byte array) Ber elements but which can have ushort or byte tags
+                        var dol = new Tag();
+                        dol.Data = new byte[1];
+                        if ((dolList[index] & 0b0001_1111) == 0b0001_1111)
+                        {
+                            dol.TagNumber = BinaryPrimitives.ReadUInt16BigEndian(dolList.AsSpan().Slice(index, 2));
+                            index += 2;
+                            dolList.AsSpan().Slice(index++, 1).CopyTo(dol.Data);
+                        }
+                        else
+                        {
+                            dol.TagNumber = dolList[index++];
+                            dolList.AsSpan().Slice(index++, 1).CopyTo(dol.Data);
+                        }
+                        dol.Parent = tag.TagNumber;
+                        tag.Tags.Add(dol);
+                    }
+                }
+                tag.Parent = parent;
+                theTags.Add(tag);
+            }
+
+        }
+
+        /// <summary>
+        /// Gather all the public information present in the credit card.
+        /// Fill then Tag list with all the found information
+        /// </summary>
+        public void FillCreditCardInformation()
```

Comment:
summary should have info where is this info available. Also I'd replace `Fill` with `Read`